src/search.cpp

```diff
@@ -696,7 +698,30 @@ Value Search::Worker::search(
         // Partial workaround for the graph history interaction problem
         // For high rule50 counts don't produce transposition table cutoffs.
         if (pos.rule50_count() < 90)
-            return ttData.value;
+        {
+            if (depth >= 8 && ttData.move && pos.legal(ttData.move) && pos.pseudo_legal(ttData.move)
```

Comment:
can you swap legal and pseudo legal checks here ? since pseudo legal is a requirement of legal

Reply:
@Disservin I agree it's better to call pseudo_legal() first.  However, I just checked and there are current instances in master when legal() is called when pseudo_legal() is false.  Here https://github.com/official-stockfish/Stockfish/blob/c4e2479a75ae34a620f68cfb686d04ce36709ab6/src/search.cpp#L1635 for example.

Reply:
See https://tests.stockfishchess.org/tests/view/682a8e256ec7634154f9a791

Reply:
The best way to explain may be that if you put an ```assert(pseudo_legal(m));``` at the top of the implementation of legal() it will trigger.  The places that call it and trigger are those I modified in the test https://tests.stockfishchess.org/tests/view/682a8e256ec7634154f9a791